server/common/check.go

```diff
@@ -17,10 +17,10 @@ func CanWrite(meta *model.Meta, path string) bool {
 
 func CanAccess(user *model.User, meta *model.Meta, reqPath string, password string) bool {
 	// if the reqPath is in hide (only can check the nearest meta) and user can't see hides, can't access
-	if meta != nil && !user.CanSeeHides() && meta.Hide != "" {
+	if meta != nil && !user.CanSeeHides() && meta.Hide != "" && !utils.PathEqual(meta.Path, reqPath) {
 		for _, hide := range strings.Split(meta.Hide, "\n") {
 			re := regexp.MustCompile(hide)
-			if re.MatchString(reqPath[len(meta.Path):]) {
+			if re.MatchString(reqPath[len(meta.Path)+1:]) {
```

Comment:
Not suitable for the root directory?

Reply:
`!utils.PathEqual(meta.Path, reqPath) `
It has been determined whether the meta directory and the directory are the same. As for `PathEqual`, I was trying to determine if `meta.Path` and `reqPath` were the same, and I was going to use `!=`, but I saw the `PathEqual` and wondered if it was the right one to use.

Access to the root directory without parent directory meta, root directory meta is the nearest meta, this code will not be run.
And the root directory can't hidden.

The hidden meta of the root directory can be set to `.*`, this will hide all files in the directory and there are no accidental passwords.

Reply:
The previous problem is that the directory hidden meta will accidentally match the current directory, so when the current directory has a meta, we do not need to strengthen the security control, and can avoid this error.

Removing the slash is to solve the problem that the directory can still be opened after hiding. The root directory does not need to be considered because it is excluded by if.

I referenced https://github.com/alist-org/alist/issues/2532

Reply:
Feel sorry I misread the scope